packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.spec.tsx

```diff
@@ -367,6 +367,22 @@ describe("horizontal overflow", (): void => {
 
         expect(renderedWithImages.instance().state.itemsHeight).toBe(0);
     });
+    test("should call `onHorizontalOverflowChange` on render if prop exists", () => {
+        const onHorizontalOverflowChangeFunction: any = jest.fn();
+        const renderedWithImages: any = mount(
+            <HorizontalOverflow
+                onHorizontalOverflowChange={onHorizontalOverflowChangeFunction}
+                className="foo"
+                managedClasses={managedClasses}
+            >
+                {imageSet1}
+            </HorizontalOverflow>
+        );
+
+        renderedWithImages.find("div.foo").simulate("load", {});
+
+        expect(onHorizontalOverflowChangeFunction).toHaveBeenCalledTimes(1);
```

Comment:
So I think this is actually the wrong way to do things. The `onHorizontalOverflowChange` should fire when the overflow has changed, same as when the `onScrollChange` is fired. However it is arbitrarily getting hit whenever the overflow *might* have changed. I think this is something that should be kept as a private variable to be updated and checked against, so when event like load etc may trigger a change in overflow, it checks first before firing the callback.

Reply:
I've updated.